src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -303,6 +299,26 @@ - (void)_handleSubscriptionReset
     os_unfair_lock_unlock(&self->_lock);
 }
 
+- (void)_handleUnsolicitedCommunication
+{
+    os_unfair_lock_lock(&self->_lock);
+
+    [self _changeState:MTRDeviceStateReachable];
+
+    id<MTRDeviceDelegate> delegate = _weakDelegate.strongObject;
+    if (delegate && [delegate respondsToSelector:@selector(deviceIsCommunicating:)]) {
+        dispatch_async(_delegateQueue, ^{
+            [delegate deviceIsCommunicating:self];
+        });
+    }
+
+    // in case this is called dyring exponential back off of subscription
+    // reestablishment, this starts the attempt right away
+    [self _setupSubscription];
```

Comment:
This works if we don't have a ReadClient set up at all.  If we do have one, this is not going to help, right?

I assume there will be follow-ups to make that better?  Is there a plan to add an API on ReadClient to trigger an immediate resubscribe attempt as needed?

Reply:
If there is already a ReadClient set up, then this is a no-op. `_setupSubscription` sets `_subscriptionActive` when that is the case.

Reply:
And yes, there will be a follow up to poke Subscriptions in the `InteractionModelEngine` layer.

Reply:
Sounds great.